### PR TITLE
Retry ssl connection errors

### PR DESCRIPTION
**Pull request by dyladan**

### CODE REVIEW

File: lib/logstash/outputs/dynatrace.rb [hunk 1]
```diff
@@ -17,6 +17,7 @@
 require 'logstash/namespace'
 require 'logstash/outputs/base'
 require 'logstash/json'
+require 'openssl'
 
 MAX_RETRIES = 5
 PLUGIN_VERSION = '0.3.2'
```
Comment (arminru) [inline lib/logstash/outputs/dynatrace.rb:23]:
```suggestion
PLUGIN_VERSION = '0.4.0'
```

File: lib/logstash/outputs/dynatrace.rb [hunk 2]
```diff
@@ -108,7 +112,6 @@ def multi_receive(events)
           end
         rescue StandardError => e
           @logger.error("Unknown error raised", :error => e.inspect)
-          raise e
```
Comment (arminru) [inline lib/logstash/outputs/dynatrace.rb:111]: Shouldn't the extra clause for well-known errors above be sufficient while unknown errors could still be rethrown?

File: lib/logstash/outputs/dynatrace.rb (hunk 1 above)
Comment (dyladan) [reply lib/logstash/outputs/dynatrace.rb:23]: Minor? This isn't adding a new feature but fixing a bug. I'd go with patch

File: lib/logstash/outputs/dynatrace.rb (hunk 2 above)
Comment (dyladan) [reply lib/logstash/outputs/dynatrace.rb:111]: If we throw it crashes the whole process which is basically never wanted unless there is an actual code error being run into.

Comment (dyladan) [reply lib/logstash/outputs/dynatrace.rb:111]: Actually I think you're right we do want to throw when we don't know what's going on

File: lib/logstash/outputs/dynatrace.rb (hunk 1 above)
Comment (arminru) [reply lib/logstash/outputs/dynatrace.rb:23]: The explicit single concurrency and adapted error handling might still change behavior for someone relying on it.

**Review by arminru — APPROVED** [review]

**Review by arminru — APPROVED** [review]

**Pull request merged**